includes/class-newspack-popups-criteria.php

```diff
@@ -0,0 +1,237 @@
+				'category' => 'reader_activity',
+				'options'  => [
+					[
+						'name'  => __( 'All users', 'newspack-popups' ),
+						'value' => 0,
+					],
+					[
+						'name'  => __( 'Has user account', 'newspack-popups' ),
+						'value' => 1,
+					],
+					[
+						'name'  => __( 'Does not have user account', 'newspack-popups' ),
+						'value' => 2,
+					],
+				],
+			],
+			/**
+			 * Referrer Sources.
+			 */
+			'sources_to_match'         => [
+				'name'              => __( 'Sources to match', 'newspack-popups' ),
+				'help'              => __( 'Segment based on traffic source', 'newspack-popups' ),
+				'description'       => __( 'A comma-separated list of domains.', 'newspack-popups' ),
+				'category'          => 'referrer_sources',
+				'matching_function' => 'list__in',
+			],
+			'sources_to_exclude'       => [
+				'name'              => __( 'Sources to exclude', 'newspack-popups' ),
+				'help'              => __( 'Segment based on traffic source - hide campaigns for visitors coming from specific sources.', 'newspack-popups' ),
+				'description'       => __( 'A comma-separated list of domains.', 'newspack-popups' ),
+				'category'          => 'referrer_sources',
+				'matching_function' => 'list__not_in',
+			],
+		];
+		foreach ( $criteria as $id => $config ) {
+			self::register_criteria( $id, $config );
+		}
+	}
+
+	/**
+	 * Enqueue scripts.
+	 */
+	public static function enqueue_scripts() {
+		wp_enqueue_script(
+			self::SCRIPT_HANDLE,
+			plugins_url( '../dist/criteria.js', __FILE__ ),
+			[],
+			filemtime( dirname( NEWSPACK_POPUPS_PLUGIN_FILE ) . '/dist/criteria.js' ),
+			true
+		);
+		wp_script_add_data( self::SCRIPT_HANDLE, 'defer', true );
+		wp_localize_script(
+			self::SCRIPT_HANDLE,
+			'newspackPopupsCriteria',
+			[
+				'config' => self::get_criteria_config(),
+			]
+		);
+		wp_enqueue_script(
+			'newspack-popups-default-criteria',
+			plugins_url( '../dist/defaultCriteria.js', __FILE__ ),
+			[ self::SCRIPT_HANDLE ],
+			filemtime( dirname( NEWSPACK_POPUPS_PLUGIN_FILE ) . '/dist/defaultCriteria.js' ),
+			true
+		);
+		wp_script_add_data( 'newspack-popups-default-criteria', 'defer', true );
+		wp_enqueue_script(
+			'newspack-popups-segments-example',
+			plugins_url( '../dist/segmentsExample.js', __FILE__ ),
+			[ self::SCRIPT_HANDLE ],
+			filemtime( dirname( NEWSPACK_POPUPS_PLUGIN_FILE ) . '/dist/segmentsExample.js' ),
+			true
+		);
+		wp_script_add_data( 'newspack-popups-segments-example', 'defer', true );
+	}
+
+	/**
+	 * Get registered criteria.
+	 *
+	 * @return array
+	 */
+	public static function get_registered_criteria() {
+		return self::$registered_criteria;
+	}
+
+	/**
+	 * Get registered criteria config to be used in the front-end.
+	 *
+	 * @return array
+	 */
+	public static function get_criteria_config() {
+		$config = [];
+		foreach ( self::$registered_criteria as $id => $criteria ) {
+			$config[ $id ] = [
+				'matchingFunction'  => $criteria['matching_function'],
```

Comment:
doesn't this throw a warning for the criteria without these attributes?

Reply:
Not by design. The default matching attribute is the criteria ID and the default matching function is `default`.

Reply:
I see it now!